client/boot/app.js

```diff
@@ -0,0 +1,56 @@
+// Initialize localStorage polyfill before any dependencies are loaded
+require( 'lib/local-storage' )();
+if ( process.env.NODE_ENV === 'development' ) {
+	require( 'lib/wrap-es6-functions' )();
+}
+
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import { invoke } from 'lodash';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import {
+	configureReduxStore,
+	locales,
+	setupMiddlewares,
+	utils
+} from './common';
+import createReduxStoreFromPersistedInitialState from 'state/initial-state';
+import detectHistoryNavigation from 'lib/detect-history-navigation';
+import userFactory from 'lib/user';
+
+const debug = debugFactory( 'calypso' );
+
+const boot = currentUser => {
+	debug( "Starting Calypso. Let's do this." );
+
+	const project = require( `./project/${ config( 'project' ) }` );
```

Comment:
I was thinking of suggesting something similar, using `process.env.PROJECT_NAME` but that felt hacky... `define-plugin` looks much cleaner and a better fit generally :)
My only concern is that we do have files that are bundled with webpack but also run directly in node depending on the context - will `PROJECT_NAME` be undefined when we run it outside of a bundle? (If so I'm sure theres a simple enough solution).

If my understanding is correct we would still need to handle the requires a little differently:

```js
const project = require( `./project/${ PROJECT_NAME }` );
// I think webpack will still treat this as 'dynamic' in that there is some logic in the require.

const projectPath =  `./project/${ config( 'project' ) }`;
const project = require( projectPath );
// might be better.
```

That said, I'm really not sure if Webpack treats the interpolated string in the same way as a concatenated string in this context - there might be some caveats.